### DISCUSSION TITLE
nRF5340 Low Power Mode

### DISCUSSION THREAD

**Discussion by krisnaresi89** [zephyrproject-rtos/zephyr#75230] [Q&A] [answered]
Hi there,

I have project that require low power mode using nRF5340. I found a very comprehensive thread [here](https://devzone.nordicsemi.com/nordic/nordic-blog/b/blog/posts/optimizing-power-on-nrf53-designs) related to low power for nRF5340.

FYI, right now I use Zephyr v3.6.0. When I compile my source code, I start to realize `CONFIG_PM` cannot be set because it depends on `HAS_PM`, and surprisingly nRF3240 doesn't have that. I think the thread above is using Zephyr v3.4.0 or below because at that version, `CONFIG_PM` doesn't have `HAS_PM` dependency.

Now, I want to implement custom idle thread hook using `pm_state_set` and `pm_state_exit_post_ops`, unfortunately I cannot do that because `CONFIG_PM` cannot be set. Is there anything we can do other that rollback to 3.4.0 and still stay on 3.6.0? Thank you for your time and assistance.

**Comment by jakubtopic** [comment, marked as answer]
Hi @krisnaresi89,

afaik Nordic does not currently implement the [System Power Management](https://docs.zephyrproject.org/latest/services/pm/system.html) on its platforms.  There is only [Device Power Management](https://docs.zephyrproject.org/latest/services/pm/device.html) subsystem available for peripheral devices.

Nordic SoCs have automatic power management when running idle (ie. `k_cpu_idle()`) and there is one extra state – System OFF – that can be entered by calling [`sys_poweroff()`](https://docs.zephyrproject.org/latest/doxygen/html/group__sys__poweroff.html#gaf4552a8ea40b4adb94f85b0a1ff06e39).

**Reply by krisnaresi89** [reply]
Hi @jakubtopic,

Thank you for your answer. Your explanation is clear so I mark it as answer. Thanks.

**Reply by e-deck** [reply]
So much head scratching. Good to know it's just not supported and I'm not crazy.
